content/blog/2022-03-29-unified-gc/index.md

```diff
@@ -0,0 +1,115 @@
+
+<p align="center">
+<img src="def_rc_partial_tracing.png" alt="alt_text" title="image_tooltip" style="zoom:50%;" />
+</p>
+
+## Generational Garbage Collection Hybrids
+
+Generational garbage collection is another well researched method of garbage collection. These collectors operate on the assumption that most objects die young. In other words, objects that have been recently allocated have a higher chance of becoming unreachable compared to objects that have been reachable for a long time. To accomplish this the heap is split into two regions: a nursery and a mature space. Objects are allocated into the nursery and at some point objects that survive long enough are moved to the mature space.
+
+The paper presents and compares three generational garbage collector algorithms: tracing generational collection, generational with a reference counted nursery, and generational with a reference counted heap. In a tracing generational collector the roots into the nursery use tracing, the roots into the mature space use tracing, the nursery uses tracing within itself, the mature space uses tracing within itself, and objects in the mature space that reference objects in the nursery use reference counting. In a generational collector with reference counted nursery, we use the same formulation except that the nursery uses reference counting instead of tracing. The advantage of this collector is that cyclic garbage will eventually be collected because the mature space is traced; the disadvantage is that it reference counts young objects, which most likely have a high mutation rate, leading to expensive write barrier operations being performed most frequently. In a generational collector with a reference counted heap, we also use the same formulation except that the mature space is reference counted instead of traced. This has the advantage that mutations in the nursery are not recorded by the write barrier which is less expensive, but the disadvantage that some additional cycle collection mechanism is required for the mature space. The relationship between these collectors is depicted below. It is important to recognize that modifications or different versions of collectors yield different design tradeoffs because they are employing tracing or reference counting differently.
+
+<p align="center">
+<img src="GCC.png" alt="alt_text" title="image_tooltip" style="zoom:25%;" />
+</p>
+
+## Cycle Collection
+
+One of the key disadvantages of reference counting is the lack of cycle collection. The authors propose two solutions, one is to run reference counting without cycle collection, and occasionally run a tracing algorithm to detect cycles. The other solution proposed was reference counting with trial deletion. In this algorithm, objects with suspiciouslly high references are traced to check for cycles.
```

Comment:
"The authors propose two solutions, one…" is a comma splice. Maybe replace the `,` with a `:`? Or just start a new sentence there.